Update docs related to Workspace tab (was "Filesystem" tab) (#3400)
MicrosoftDocs/edge-developer: microsoft-edge/devtools-guide-chromium/quick-source/quick-source-tool.md

ms.topic: conceptual
ms.service: microsoft-edge
ms.subservice: devtools
ms.date: 03/07/2025
---
# Display or edit source files using the Quick source tool
<!-- https://developer.chrome.com/docs/devtools/quick-source -->
<!-- first give the always-available GUI way, then keyboard shortcuts, then transient UI in instructions that aren't present when a file is open -->

> Use the **Quick source** tool to display or edit source files when using a tool other than the **Sources** tool.

Author: ```suggestion
Use the **Quick source** tool to display or edit source files at the same time as displaying a tool other than the **Sources** tool.
```

![The Quick source tool](./quick-source-tool-images/quick-source-tool.png)

The main place to view source files in the DevTools is within the **Sources** tool.  But sometimes you need to access other tools, such as **Elements** or **Console**, while viewing or editing your source files.  Use the **Quick source** tool, which by default opens in the **Quick View** panel at the bottom of DevTools.

<!-- ------------------------------ -->
#### Use the Quick source tool in conjunction with the Sources tool

To display the **Quick source** tool in **Quick View** at the bottom of DevTools, a tool other than the **Sources** tool must be open in the **Activity Bar** at the top of DevTools.

The **Quick source** tool provides an alternative display of the files that are open in the **Sources** tool.  The **Quick source** tool has a subset of features.  For full features, temporarily switch to the **Sources** tool.

If you open or close a file in one tool, the file is opened or closed in the other tool.  In both tools, there's a tab for each open file.

<!-- ====================================================================== -->
## Open a file in the Quick source tool

> To make a file appear in the **Quick source** tool:

Author: ```suggestion
To make a file appear in the **Quick source** tool, open a file in the **Sources** tool, as follows:
```

   **Open a file by using the Sources tool:**

1. In the **Activity Bar** at the top of DevTools, select the **Sources** (![The Sources tool icon](./quick-source-tool-images/sources-icon.png)) tool.

   The **Quick View** pane is minimized at the bottom of DevTools.

1. In the **Sources** tool, in the **Page** tab, select one or more files, such as `to-do.js`.

   The tabs for the selected files are opened in the **Sources** tool:

   ![The Sources tool > Page tab with a file selected](./quick-source-tool-images/sources-tool-page-tab-file-selected.png)

1. In the **Activity Bar** at the top of DevTools, select a tool other than **Sources**, such as **Elements**.

   The **Quick source** tool expands in **Quick View** at the bottom of DevTools, and has a tab for each open file:

   ![A file displayed in the Quick source tool.](./quick-source-tool-images/file-displayed-in-quick-source.png)<!-- 1st use -->

> 

Author: ```suggestion

You can now view or edit the file in the **Quick source** tool, at the same time as displaying a different tool.

```

   The **Quick View** pane is minimized at the bottom of DevTools.

1. In the **Sources** tool, in the **Page** tab, select one or more files, such as `to-do.js`.

   The tabs for the selected files are opened in the **Sources** tool:

   ![The Sources tool > Page tab with a file selected](./quick-source-tool-images/sources-tool-page-tab-file-selected.png)

1. In the **Activity Bar** at the top of DevTools, select a tool other than **Sources**, such as **Elements**.

   The **Quick source** tool expands in **Quick View** at the bottom of DevTools, and has a tab for each open file:

   ![A file displayed in the Quick source tool.](./quick-source-tool-images/file-displayed-in-quick-source.png)<!-- 1st use -->

You can now view or edit the file in the **Quick source** tool, at the same time as displaying a different tool.

See also:
* [Using the Page tab to explore resources that construct the current webpage](../sources/index.md#using-the-page-tab-to-explore-resources-that-construct-the-current-webpage) in _Sources tool overview_.

<!-- ====================================================================== -->
## Edit local source files in DevTools by selecting a Workspace folder

> By selecting a local folder as a **Workspace**, you can make edits in DevTools **Quick source** tool (or **Sources** tool) and have those edits be saved to your local drive.  This way, when you edit the folder's files in DevTools, the edits are not only made to the transient file that's returned by the server, but are also saved to the source files on your local drive.

Author: ```suggestion
By selecting a local folder as a **Workspace**, you can edit files from within the **Quick source** tool at the same time as displaying a different tool (or from within the **Sources** tool), and have those edits be saved to your local drive.  This way, when you edit the folder's files in DevTools, the edits are not only made to the transient file that's returned by the server, but are also saved to the source files on your local drive.
```

To open a server webpage file in the **Quick source** tool and save any edits to your local **Workspace** folder of source files:

1. Open a webpage that's on a web server; for example, open [Demo To Do](https://microsoftedge.github.io/Demos/demo-to-do/) in a new window or page.  That page is hosted on the web server `microsoftedge.github.io`.

   **Open the DevTools Quick source tool:**

1. Right-click the webpage and then select **Inspect**.

   DevTools opens.

1. If the **Quick View** toolbar isn't shown at the bottom of DevTools, press **Esc**.

1. In the **Quick View** toolbar at the bottom of DevTools, click the **More tools** button, and then select **Quick source**.

   The **Quick source** tool opens in **Quick View** at the bottom of DevTools:

   ![The Quick source tool, containing a Select folder link](./quick-source-tool-images/quick-source-select-folder.png)

   The **Quick source** tool might show instructions including an **Select folder** link, or it might display a file.  Instead of relying on that transient link, the steps further below show how to use the **Workspace** tab in the **Sources** tool, to select a local folder of source files.

>    **Have local source files that match the web server files:**

Author: create sep. h2 Clone repo
